src/pretty.jl

```diff
@@ -8,23 +8,51 @@ macro pretty(json)
 end
 
 """
-    JSON3.pretty(x; kw...)
-    JSON3.pretty(io, x; kw...)
+    JSON3.AlignmentContext(alignment=:Left, indent=4, level=0, offset=0)
+
+Specifies the indentation of a pretty JSON string.
+
+## Keyword Args
+* `alignment`: A Symbol specifying the alignment type. Can be `:Left` to left-align
+               everything or `:Colon` to align at the `:`.
+* `indent`: The number of spaces to indent each new level with.
+* `level`: The indentation level.
+* `offset`: The indentation offset.
+"""
+Base.@kwdef mutable struct AlignmentContext
+    alignment::Symbol = :Left
+    indent::UInt16 = 4
+    level::UInt16 = 0
+    offset::UInt16 = 0
+    function AlignmentContext(alignment, indent, level, offset)
+        if alignment != :Left && alignment != :Colon
+            throw(ArgumentError("Alignment :$(alignment) is not supported. \
```

Comment:
I think the trailing `\` character on this line is causing some tests to fail

Reply:
I changed that line to something more 1.0 friendly. Hopefully the tests pass now!